pkg/rbac/parser.go

```diff
@@ -44,8 +48,62 @@ type Rule struct {
 	URLs      []string `marker:"urls,optional"`
 }
 
+// target generates a string representing the resources and/or non-resources a Rule applies.
+func (r *Rule) target() string {
+	r.cleanUp()
+	s := []string{
+		strings.Join(r.Groups, ","),
+		strings.Join(r.Resources, ","),
+		strings.Join(r.URLs, ","),
+	}
+	return strings.Join(s, ",")
+}
+
+// addVerbs adds new verbs into a Rule
+func (r *Rule) addVerbs(verbs []string) {
+	r.Verbs = removeDup(append(r.Verbs, verbs...))
+}
+
+// cleanUp removes duplicates from each field of a Rule, and sorts each field.
+func (r *Rule) cleanUp() {
```

Comment:
nit: cleanup is insufficiently descriptive.  Maybe `normalize` or something?

Reply:
`normalize` is definitely better. That word didn't come to me yesterday!
Fixed it